src/game.h

```diff
@@ -9,33 +9,40 @@ class Game
     Game();
     ~Game();
 
-    void init(std::string build_path);
+    // This class is not ought to be copied or assigned!
+    Game(const Game&) = delete;
+    Game(Game&&) = delete;
+    Game& operator= (const Game&) = delete;
+    Game& operator= (Game&&) = delete;
 
-    void draw();
+    bool init(const std::string& build_path);
+
+    void draw() const;
     void handleInput();
-  
+
     void moveBlockDown();
 
     bool gameOver = false;
     int score = 0;
     Music music;
 
-private:
+private: // methods
     void moveBlockLeft();
     void moveBlockRight();
-    Block getRandomBlock();
-    std::vector<Block> getAllBlocks();
-    bool isBlockOutside();
+    const Block& getRandomBlock() const;
```

Comment:
Since allBlocks are now stored in a singleton we may now return a const Reference! So the CPU does not longer need to perform memory copy ops and may just use C++References (that are actually pointers).